src/js/common/schemaform/Address.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import _ from 'lodash';
+import { set, assign } from 'lodash/fp';
+
+import { states } from '../utils/options-for-select';
+
+/**
+ * Input component for an address.
+ *
+ * No validation is provided through a currently stubbed isAddressValid function.
+ */
+class Address extends React.Component {
+  constructor(props) {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.onPropertyBlur = this.onPropertyBlur.bind(this);
+    this.isRequired = this.isRequired.bind(this);
+    this.state = { value: props.formData, touched: { street: false, city: false, state: false, country: false, postalCode: false } };
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('address-input-');
+  }
+
+  onPropertyBlur(name) {
+    return (path = []) => {
+      this.props.onBlur([name].concat(path));
+    };
+  }
+
+  handleChange(path, update) {
+    let newState = set(path, update, this.props.formData);
+
+    // if country is changing we should clear the state
+    if (path === 'country') {
+      newState = set('state', undefined, newState);
+    }
+
+    this.props.onChange(newState);
+  }
+
+  isRequired(name) {
+    const { schema } = this.props;
+    const schemaRequired = Array.isArray(schema.required) &&
+      schema.required.indexOf(name) !== -1;
+
+    if (schemaRequired) {
```

Comment:
I think we can remove this if statement and just return `schemaRequired` at the bottom.